client/packages/common/src/hooks/useBreadcrumbs/useBreadcrumbs.tsx

```diff
@@ -11,49 +11,65 @@ export interface UrlPart {
 }
 
 type BreadcrumbState = {
-  setSuffix: (suffix?: string | React.ReactElement) => void;
-  suffix?: string | React.ReactElement;
+  setBreadcrumbRenderers: (
+    renderers?: Record<number, (part: UrlPart) => string | React.ReactElement>
```

Comment:
```suggestion
    renderers?: Record<part: string, (part: UrlPart) => string | React.ReactElement>
```
Could we do something like this?

So that you call it with...
`setBreadcrumbRenderers({ data?.id: () => data?.assetNumber ?? '' });`
or
`setBreadcrumbRenderers({ 'assets': () => t('label.assets')});

To me it's not intuative where index starts? Seems like it's an index from the end currently?
So we want to use 1 for the last element, 2 for the second to last?

I'd be tempted to just accept reactNode rather than renderer function too?

Depends if we want a single render which reacts to different values, but if we'd still need to specify all the indexes currently already, were you imagining passing in a render that would take an ID, so an API look up to determine the right label for that element?

Maybe something like this would be nicer? 
`setBreadcrumbRenderers({ data?.id: data?.assetNumber ?? '' });`
`setCustomBreadcrumbs({ 'assets': t('label.assets')});`

Reply:
yeah, that's nice!
personally, I prefer thinking in terms of indices rather than static values - so I would keep with the current implementation. I also like the suggestion to use a react node rather than a renderer, though if you do that you lose the ability to respond to a part, actually fine in practice as we have no need for this; all the same its kinda neat to have a function which takes the url part and provides an appropriate response. that's more flexible.

Reply:
Thanks for input! Yeah I like the just providing a value rather than a function, that should make it a little less unwieldy. 

Yeah I thought accepting the part and being able to do something with it would be helpful, but I think in any case where we want that, we still have access to the URL parts, can do the logic and then pass that into `setCustomBreadcrumbs` 👀 

I think I'll stick with index - potentially undefined object keys feels like asking for trouble in JS 🤣 Though I do like how easy it is to tell what you are replacing!